homeassistant/components/binary_sensor/trend.py

```diff
@@ -0,0 +1,141 @@
+    vol.Optional(ATTR_FRIENDLY_NAME): cv.string,
+    vol.Optional(CONF_INVERT, default=False): cv.boolean,
+    vol.Optional(CONF_SENSOR_CLASS, default=None): SENSOR_CLASSES_SCHEMA
+
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SENSORS): vol.Schema({cv.slug: SENSOR_SCHEMA}),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the template sensors."""
+    sensors = []
+
+    for device, device_config in config[CONF_SENSORS].items():
+        entity_id = device_config[ATTR_ENTITY_ID]
+        attribute = device_config.get(CONF_ATTRIBUTE)
+        friendly_name = device_config.get(ATTR_FRIENDLY_NAME, device)
+        sensor_class = device_config[CONF_SENSOR_CLASS]
+        invert = device_config[CONF_INVERT]
+
+        sensors.append(
+            SensorTrend(
+                hass,
+                device,
+                friendly_name,
+                entity_id,
+                attribute,
+                sensor_class,
+                invert)
+            )
+    if not sensors:
+        _LOGGER.error("No sensors added")
+        return False
+    add_devices(sensors)
+    return True
+
+
+class SensorTrend(BinarySensorDevice):
+    """Representation of a Template Sensor."""
+
+    # pylint: disable=too-many-arguments, too-many-instance-attributes
+    def __init__(self, hass, device_id, friendly_name,
+                 target_entity, attribute, sensor_class, invert):
+        """Initialize the sensor."""
+        self._hass = hass
+        self.entity_id = generate_entity_id(ENTITY_ID_FORMAT, device_id,
+                                            hass=hass)
+        self._name = friendly_name
+        self._target_entity = target_entity
+        self._attribute = attribute
+        self._sensor_class = sensor_class
+        self._invert = invert
+        self._state = None
+        self.from_state = None
+        self.to_state = None
+
+        self.update()
+
+        def template_sensor_state_listener(entity, old_state, new_state):
+            """Called when the target device changes state."""
+            self.from_state = old_state
+            self.to_state = new_state
+            self.update_ha_state(True)
+
+        track_state_change(hass, target_entity,
+                           template_sensor_state_listener)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._state
+
+    @property
+    def sensor_class(self):
+        """Return the sensor class of the sensor."""
+        return self._sensor_class
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    def update(self):
+        """Get the latest data and update the states."""
+        if (self.from_state and self.to_state and
```

Comment:
Oh, and you're storing state objects, not actual states in `to` and `from` so it will never be `'Unknown'`